examples/spells/BoundedUInt.qnt

```diff
@@ -0,0 +1,607 @@
+module BoundedUInt {
+  /// The size of this integer type in bits.
+  const BITS: int
+
+  /// The smallest value that can be represented by this integer type.
+  pure val MIN = 0
+
+  /// The largest value that can be represented by this integer type.
+  pure val MAX = (2^BITS) - 1
+  
+  /// Record-representation of an unsigned bounded integer.
+  /// If the `error` field is nonempty, the record represents an exception of some
+  /// sort happening during computation, such as an overflow.
+  /// Otherwise, the record represents the integer `v`, such that `MIN <= v <= MAX`. 
+  type T = { v: int, error: str }
```

Comment:
I'm wondering if actual use case needs an `error` message string. Would it be possible that having `error: bool` would be more ergonomic and just knowing that there is an error one way or the other would suffice?

Reply:
I can rename it `UintT` (`UInt` clashes with the constructor). As for the error field, I prefer keeping it. You can always call `i.isValid`, and get a boolean evaluation, whereas the error string is more descriptive (and already implemented)

Reply:
The constructor could be `intToUInt`. I think this is how we are making converters of, e.g., maps to sets: https://uxdesign.cc/designing-for-delight-is-dead-677bca6aebd1

Reply:
Miss pasted! Sorry. That should be https://github.com/informalsystems/quint/blob/main/quint/src/builtin.qnt#L310

Reply:
> I'm wondering if actual use case needs an `error` message string. Would it be possible that having `error: bool` would be more ergonomic and just knowing that there is an error one way or the other would suffice?

The `error` field is similar to what was done with the modelling of Decimals in https://github.com/informalsystems/quint-sandbox/blob/main/decimal/decimal.qnt. While a Boolean would suffice, having a string is almost as cheap (from the analysis p.o.v.) as having a Boolean, and it it produces better counterexamples.

Reply:
I'm fine with leaving it as an error message if that has proven useful.

I think the remaining item left for this comment is then the type name.

Reply:
I think 
```
UInt: int => UIntT
```
looks much nicer than
```
intToUInt: int => UInt
```
mainly because, if I have to type `intToUInt` for every integer literal in my spec, that's a lot of line-width bloat.
@konnov @thpani @bugarela

Reply:
True, it looks nicer. If we go with an interface that requires wrapping values, makes sense to keep it short.